cstar/tests/unit_tests/base/test_input_dataset.py

```diff
@@ -54,7 +54,7 @@ def local_input_dataset():
         mock_basename.return_value = "local_file.nc"
 
         dataset = MockInputDataset(
-            location="some/local/source/path/local_file.nc",
+            location=Path("some/local/source/path/local_file.nc"),
```

Comment:
This appears to be a non-typesafe change. `InputDataset__init__` signature is:

```python
    def __init__(
        self,
        location: str,
        file_hash: Optional[str] = None,
        start_date: Optional[str | dt.datetime] = None,
        end_date: Optional[str | dt.datetime] = None,
    ):
```

Sending a path here should be avoided. It creates an unintended dependency on the implementation and this test.

Consider adding full type hinting to your other tests to catch this.